assets/js/paper-downloads.js

```diff
@@ -0,0 +1,494 @@
+     * Parse download count from HTML using regex
+     */
+    parseDownloadCount(html) {
+        try {
+            const match = html.match(this.downloadRegex);
+            if (match && match[1]) {
+                return parseInt(match[1], 10);
+            }
+            return null;
+        } catch (error) {
+            console.warn('Error parsing download count:', error);
+            return null;
+        }
+    }
+
+    /**
+     * Display the download count with animation
+     */
+    displayCount(count, timestamp, isFromCache) {
+        this.currentCount = count;
+        this.lastUpdated = timestamp;
+
+        const countElement = Utils.DOM.select('#download-count');
+        const lastUpdatedElement = Utils.DOM.select('#last-updated');
+
+        if (countElement) {
+            // Animate counter from current to new value
+            this.animateCounter(countElement, count);
+        }
+
+        if (lastUpdatedElement) {
+            const timeStr = new Date(timestamp).toLocaleString('zh-TW');
+            const cacheIndicator = isFromCache ? ' (快取)' : '';
+            lastUpdatedElement.textContent = `最後更新：${timeStr}${cacheIndicator}`;
+        }
+    }
+
+    /**
+     * Animate counter with counting effect
+     */
+    animateCounter(element, targetCount) {
+        const startCount = this.currentCount || 0;
+        const duration = 2000; // 2 seconds
+        const startTime = performance.now();
+
+        const animate = (currentTime) => {
+            const elapsed = currentTime - startTime;
+            const progress = Math.min(elapsed / duration, 1);
+
+            // Easing function for smooth animation
+            const easeOutQuart = 1 - Math.pow(1 - progress, 4);
+            const currentCount = Math.floor(startCount + (targetCount - startCount) * easeOutQuart);
+
+            element.innerHTML = `<span class="count-number">${currentCount.toLocaleString()}</span>`;
+
+            if (progress < 1) {
+                requestAnimationFrame(animate);
+            } else {
+                // Final value
+                element.innerHTML = `<span class="count-number">${targetCount.toLocaleString()}</span>`;
+                // Add pulse effect when done
+                element.style.animation = 'counterPulse 0.5s ease-out';
+                setTimeout(() => {
+                    element.style.animation = '';
+                }, 500);
+            }
+        };
+
+        requestAnimationFrame(animate);
+    }
+
+    /**
+     * Set loading state
+     */
+    setLoadingState(isLoading) {
+        this.isLoading = isLoading;
+        const refreshBtn = Utils.DOM.select('#refresh-downloads');
+        const refreshIcon = Utils.DOM.select('.refresh-icon');
+        const refreshText = Utils.DOM.select('.refresh-text');
+
+        if (refreshBtn) {
+            refreshBtn.disabled = isLoading;
+            if (isLoading) {
+                Utils.DOM.addClass(refreshBtn, 'opacity-50 cursor-not-allowed');
```

Comment:
Utils.DOM.addClass and only add one class at one time. Call this function multiple times if you need to add multiple classes.